tests/e2e/cluster_creation_test.go

```diff
@@ -55,4 +58,62 @@ var _ = Describe("Create cluster", func() {
 			Expect(clusterOutput.ClusterID).ToNot(BeEmpty())
 			Expect(clusterOutput.ClusterID).ToNot(Equal(originalClusterID))
 		})
+
+	It("cluster waiter should fail on error/uninstalling cluster - [id:74472]", ci.Day1Supplemental, ci.Medium,
+		func() {
+			By("Retrieve cluster with error/uninstalling status")
+			params := map[string]interface{}{
+				"search": "status.state='error' or status.state='uninstalling'",
+				"size":   -1,
+			}
+			resp, err := cms.ListClusters(ci.RHCSConnection, params)
+			Expect(err).ToNot(HaveOccurred())
+			clusters := resp.Items().Slice()
+
+			if len(clusters) <= 0 {
+				Skip("No cluster in 'error' or 'uninstalling' state to launch this case")
+			}
+			cluster := clusters[0]
```

Comment:
Can we simulate one error cluster for the testing? I already exposed the NAT gateway to make cluster installation failure

Reply:
So you mean having a creation + failure ?
That will be more complex as the `Apply` is going to wait for the cluster creation and I have no possibility to retrieve the cluster ID for now because of https://issues.redhat.com/browse/OCM-7904
But I can create a follow-up issue to improve this. wdyt ?

Reply:
Make sense.

Reply:
Created https://issues.redhat.com/browse/OCM-9239